### PR TITLE
[feat][refactor] Grid Component

### PR DESCRIPTION
**Pull request by Hamdrive**
Created the grid component for the component library, and also created the documentation site for the same.

### **For PR Review**: Please review only the files pertaining to grid, other HTML files have been updated to reflect grid option in the respective sidebar menu

Deployed here: [Grid Documentation site](https://deploy-preview-37--powertrain-ui.netlify.app/documentation/html/grid.html)

Consists of 3 sections on the doc site:
1. 2-column grid
2. 3-column grid
3. Other types of grid layouts

image

image

image

### CODE REVIEW

File: Documentation/css/grid.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@import url("../../stylesheets/sidebar.css");
+@import url("./getstarted.css");
+@import url("../../components/grid/grid.css");
+
+.example {
+  width: 100%;
+  margin: 1rem 0;
+  display: flex;
+  flex-direction: column;
+  padding: var(--size-5);
```
Comment (GuptaEesh) [inline Documentation/css/grid.css:13]: Does this variable follow the rem convention or px convention?
- If it's rem then pls change 1 rem to variable.
- If it's px then pls change 10px to variable.

**Review by GuptaEesh — COMMENTED** [review]
Apart from the suggestion added, code looks good.

File: components/grid/grid.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.grid-2 {
+  grid-template-columns: 1fr 1fr;
+}
+
+.grid-3 {
+  grid-template-columns: 1fr 1fr 1fr;
+}
+
```
Comment (lannisterrr) [inline components/grid/grid.css:13]: Code looks fine with descriptive class names.

Comment (Hamdrive) [reply components/grid/grid.css:13]: Thanks for pointing it out, have refactored to use `repeat()` property of grids

**Comment by Hamdrive** [comment]
Last push done is `[refactored] grid definitions
made changes recommended by reviewers, pushing branch to dev and closing PR

File: Documentation/css/grid.css (hunk 1 above)
Comment (Hamdrive) [reply Documentation/css/grid.css:13]: Variable is defined in root with rem units. Yes I should have used it on line 10 as well. Thanks for pointing it out!

**Pull request merged**